# Mask reference genome

Use Unfiltered vcf (the output I get from running Stacks populations with no filtering and normal ref genome)

Then use bedtools: https://bedtools.readthedocs.io/en/latest/content/tools/maskfasta.html

command:

```
module load BEDTools
```

```
bedtools maskfasta -fi mohua_ref_genome.fna -bed unfiltered.vcf -fo masked_ref_genome.fasta
```

then I have a masked .fasta file (output refgenome) -> this is my new masked reference genome
