# Mask reference genome

Use Unfiltered vcf (the output I get from running Stacks populations with no filtering and normal ref genome)

Then use bedtools: https://bedtools.readthedocs.io/en/latest/content/tools/maskfasta.html

command:

```
module load BEDTools
```

```
bedtools maskfasta -fi mohua_ref_genome.fna -bed unfiltered.vcf -fo masked_ref_genome.fasta
```

then I have a masked .fasta file (output refgenome) -> this is my new masked reference genome

Side notes: As coding is different in bedtools (e.g., in vcf first position is 1 in bedtools/python first position is 0) -> check if it masks right position:

Take unfiltered vcf look at first site and manually replace with -100 so it does not exist and I should get an error.
Next thing to do is put 0 for the first site -> I should get error again as it should expect a vcf as input and vcf does not have 0 as a position. If I donâ€™t get error it might not mark the right position
Hardmask -> replace with N (not with lowercase a/g/t/c)

Could get slightly less SNPs in the end