# Filtering

## filter overall dataset

Subsets were filtered individually.

My downstream analyses have shown correlation between inbreeding coefficient (FH) and missing data (F_MISS), to remove this correlation -> find ideal minimum depth to filter for.
This was done using a using all individuals together.

Remove any individuals with more than 70% of missing data:
Look at missing data per individual using vcftools and --missing-indv

```
module load VCFtools
```

```
vcftools --vcf unfiltered_mask.vcf --missing-indv
```

download file and sort by F_MISS (in excel), 11 individuals have more than 70% of missing data. Remove those:

```
vcftools --vcf unfiltered_mask.vcf --remove remove_11indv.txt --recode
```

rename and move file into its own directory

```
mv out.recode.vcf removed_11indv.vcf
mv removed_11indv.vcf all_minus_11indv
```


To find ideal maximum depth, use unfiltered vcf file
```
vcftools --vcf unfiltered_mask.vcf --depth
```
Look at output file, maximum depth is 22, so filter for maximum depth of 44 (double). Smallest mean depth is 2. 


Now filter for different -minDP values:

```
vcftools --vcf removed_11indv.vcf --minDP 2 --maxDP 44 --max-missing 0.8 --recode
```

```
mv out.recode.vcf filtered_DP2_44.vcf
```

```
vcftools --vcf removed_11indv.vcf --minDP 3 --maxDP 44 --max-missing 0.8 --recode
```

```
mv out.recode.vcf filtered_DP3_44.vcf
```

```
vcftools --vcf removed_11indv.vcf --minDP 4 --maxDP 44 --max-missing 0.8 --recode
```

```
mv out.recode.vcf filtered_DP4_44.vcf
```

Then get .imiss and .het report for each .vcf (code only shown for minDP 2, make sure to always rename file to not overwrite them):

```
vcftools --vcf filtered_DP2_44.vcf --missing-indv
vcftools --vcf filtered_DP2_44.vcf --het
```

From .depth and .het output for each depth create file with sample ID, F_MISS and F for analysis in R.
Looks like no significant relationship between F_MISS and F when minDP = 3.
Accordingly, filtering will be as below: 

## filter subsets

filtering followed the same process for each subset and is only shown for the 'mainland' subset here.

make the subset:

```
vcftools --vcf unfiltered_mask.vcf --keep mainland_184_keep.txt --recode
mv out.recode.vcf mainland_unfiltered.vcf
```

### filter for individuals with high % of missing data

```
vcftools --vcf mainland_unfiltered.vcf --missing-indv
```

remove individuals with missingness of 70% or more:

```
vcftools --vcf mainland_unfiltered.vcf --remove high_miss_IDs.txt --recode
mv out.recode.vcf mainland-8highmissIDs.vcf
```

I have five difference subsets:
- Mainland populations: 8 individuals removed
- Mainland populations for Fst (minus Caples individuals): 8 individuals removed
- Blue Mountains + island populations: 3 individuals removed
- Landsborough population: 1 individual removed
- Glenorchy populations (Routeburn, Dart and Caples): 2 individuals removed
- Routeburn + Dart population: 2 individuals removed

### filter for depth and missing SNPs

All subsets were filtered for a minimum depth of 3. Maximum depth was chosen by doubling the highest depth in the individual subsets (for identification of ZW scaffolds see below):


```
vcftools --vcf mainland-8highmissIDs.vcf --depth
```

Highest depth in this subset is 22, so filtering for --maxDP 44. 
Further filtering allows 20% of missing data (SNPs) and removed SNPs on scaffolds associated to the Z or W chromosome.

```
vcftools --vcf mainland-8highmissIDs.vcf --minDP 3 --maxDP 44 --max-missing 0.8 --recode
mv out.recode.vcf mainland_DP3_44_MD08_ZWremoved.vcf
```


## identifying SNPs on sex chromosomes

Create index fasta file for mohua reference genome

```
module load SAMtools
samtools faidx mohua_ref_genome.fasta
```


SNPs that were on scaffolds that are associated with the Z or W chromosome were identified by aligning scaffolds to the chromosome level assembly reference geneom of the New Caledonian crow (Accession number xxxxxx)

This process followed th pipeline described in this script by NAME, LINK HERE

Mohua scaffolds that align to the Z and W chromosome are in 'mohua_ZWscaffold_names.txt'. 
Use this file to make a .bed file:

```
awk 'BEGIN {FS="\t"}; {print $1 FS "0" FS $2}' mohua_ref_genome.fasta.fai > mohua_genome.bed
sort mohua_genome.bed >> mohua_genome.bed
join mohua_ZWscaffold_names.txt mohua_genome.bed > mohua_sex_scaffolds.bed
```

remove Z/W associated scaffolds using vcf tools, e.g.:

```
vcftools --vcf filtered_DP3_44_MD08.vcf --exclude-bed mohua_sex_scaffolds.bed --recode
```













