# Phylogenetic analysis using SNAPP

## create input file

using a haploid subset of 5000 SNPs to cope with computational demand.

make vcf2nex.pl script by creating the file (e.g., using nano) and copying the code in here.
Code can be found here: https://github.com/BEAST2-Dev/SNAPP/blob/master/script/vcf2nex.pl

make a nexus file using vcf2nex.pl (leave the <> around input filename): 

```
perl vcf2nex.pl < file.vcf > file.nex
```

to make file 'haploid' open .nex in texteditor and delete every 2nd row, adjust 'ntaxa=' accordingly' (and potentially add 'MISSING=.'). 

Open Beauti, File -> Template -> SNAPP
File -> Add Alignment and add the .nex file 
Assign populations and work through parameters, prior and MCMC setting -> save file (will be safed as .xml) 

## Run beast:

```
#!/bin/bash -e
#SBATCH --job-name=beast_job.sh # job name (shows up in the queue)
#SBATCH --time=168:00:00      # Walltime (HH:MM:SS)
#SBATCH --mem=64000MB          # Memory in MB
#SBATCH -c 16         # number of cores

pwd

module load BEAST/2.6.6
beast -threads -16 Beauti_file_haploid.xml
```

Visualize results in DensiTree