# EEMS

### Create .diffs input file for EEMS:

1. modify my normal .vcf to have all SNPs on one chromosome
	
2. convert that new .vcf to plink format using VCFtools (--plink, created .ped and .map)

```
vcftools --vcf mainland_DP3_44_MD08_ZWremoved_1Chrom.vcf --plink
```

3. use plink to make bed file out of ped and map file (--make-bed):

```
plink --file mainland --make-bed
```

4. copy bed files (.bed, .bim, .fam) to the location I want them 

5. use the bed2diffs command to make a .diffs file:

```
source ~/bootstrap_eemssnps 

bed2diffs_v1 --bfile filename
```

## The .coord file need to be like this:

168.3790567     -44.54467449
168.269704      -44.715614
168.269704      -44.715614
169.2643698     -44.15958937
...

## Run EEMS:

```
source ~/bootstrap_eemssnps 
runeems_snps --params params-chain1.ini 
```

## Plot EEMS results in R:

```{r}
library("rEEMSplots")

eems.plots(mcmcpath = 'output/mainland-chain1/', plotpath = 'output/mainland-chain1/plots/', longlat = TRUE, add.grid = TRUE, add.outline = TRUE, add.demes = TRUE)

#plot multiple:

eems.output = list.dirs("./output",recursive = FALSE)

combined.mcmcpath = eems.output

combined.plotpath = "./output/combined_results"


eems.plots(mcmcpath = combined.mcmcpath, plotpath = combined.plotpath, longlat = TRUE, add.grid = TRUE, add.outline = TRUE, add.demes = TRUE)
```

see: https://github.com/dipetkov/eems/issues/4