# Estimate relatedness using *related* package in R:

Example for Wang relatedness:

load package:

```{r}
library(related)
```

calculate relatedness:

```{r}
data <- read.table("bluemt-isl_related.ped", header=F, stringsAsFactors=F) #input is a .ped file that was created with plink using a .vcf input file

#remove column 2-6 to match the input file for coancestry function
data$V2 <- NULL
data$V3 <- NULL
data$V4 <- NULL
data$V5 <- NULL
data$V6 <- NULL

output <- coancestry(data, wang = 2)

write.table(output$relatedness, file = "wang_relatedness.txt")
```

plot relatedness:

```{r}
RelatednessData <- read.table("wang_relatedness.txt")

popmap <- read.table("popmap_bluemt_islands.txt")
levels(as.factor(popmap[,2]))


######## Anchor
inds_of_Anchor <- popmap[which(popmap[,2]=="AnchorIsland"),1]

first_subset_RelatednessData_Anchor <- RelatednessData[which(RelatednessData[,2]%in%inds_of_Anchor),]

second_subset_RelatednessData_Anchor <- first_subset_RelatednessData_Anchor[which(first_subset_RelatednessData_Anchor[,3]%in%inds_of_Anchor),]
anchor_relatedness <- second_subset_RelatednessData_Anchor

#do the above for each of the populations in the comparison 


#then add population name to individuals relatedness data frames:
anchor_relatedness$Population <- "Anchor Island"
blue_relatedness$Population <- "Blue Mountains"
breaksea_relatedness$Population <- "Breaksea Island"
chalky_relatedness$Population <- "Chalky Island"
whenuahou_relatedness$Population <- "Whenua Hou"


#combine the 5 data frames into one
relatedness <- rbind(anchor_relatedness, blue_relatedness, breaksea_relatedness, chalky_relatedness, whenuahou_relatedness)

#add additional column for estimator so i can later plot wang and trioML in one plot:
relatedness$estimator <- "wang"

# can delete unwanted columns to make df easier to overview:
relatedness$trioml <- NULL
relatedness$lynchli <- NULL
relatedness$lynchrd <- NULL
relatedness$ritland <- NULL
relatedness$quellergt <- NULL
relatedness$dyadml <- NULL 


#bring populations in right order for plotting and plot:

relatedness$Population <- factor(relatedness$Population , levels=c("Blue Mountains", "Breaksea Island", "Anchor Island", "Whenua Hou", "Chalky Island"))

boxplot(relatedness$wang ~ relatedness$Population, col="white", border= c("#71BA31", "#4FADEA", "#2F26BA", "#2F26BA", "#2F26BA"), ylab="relatedness (wang)", xlab = "", frame=F, lwd=1.5, par(cex.lab=1), par(cex.axis=1))
```

compareestimators:

```{r}
#was done with a  SNP subset

write.table(data, file = "filename_genotypedata.txt", quote = FALSE, col.names = FALSE, row.names = FALSE, sep = "\t")

genotypeData_100SNP <- readgenotypedata("filename_genotypedata.txt")
compareestimators(genotypeData_100SNP, 68)
```